allennlp/data/samplers/samplers.py

```diff
@@ -0,0 +1,263 @@
+import itertools
+from typing import List, Iterable, Tuple, Dict, cast
+import logging
+from torch.utils import data
+
+from allennlp.common.registrable import Registrable
+
+from allennlp.common.util import add_noise_to_dict_values, lazy_groups_of
+from allennlp.data.instance import Instance
+
+logger = logging.getLogger(__name__)
+
+
+class Sampler(Registrable):
+    """
+    A wrapper around the pytorch [Sampler](https://pytorch.org/docs/stable/_modules/torch/utils/data/sampler.html)
+    which allows us to register it with `Registrable.`
+    """
+
+    def __iter__(self) -> Iterable[int]:
+
+        raise NotImplementedError
+
+
+class BatchSampler(Registrable):
+    """
+    A wrapper around the pytorch
+    [BatchSampler](https://pytorch.org/docs/stable/data.html#torch.utils.data.BatchSampler)
+    which allows us to register it with `Registrable.`
+    """
+
+    def __iter__(self) -> Iterable[List[int]]:
+
+        raise NotImplementedError
+
+
+@Sampler.register("sequential")
+class SequentialSampler(Sampler, data.SequentialSampler):
+    def __init__(self, data_source: data.Dataset, **kwargs):
```

Comment:
I did this because I thought that the way we construct samplers below, where we always pass a `data_source` argument even if the specific class doesn't take that argument, requires it? E.g we can't know apriori if a sampler takes a data_source or not.

Reply:
No, the `from_params` code figures that out. You can safely remove the `kwargs`.